controller/src/main/java/io/pravega/controller/store/stream/PersistentStreamBase.java

```diff
@@ -1118,10 +1118,13 @@ private boolean isStreamCutValidInternal(Map<Long, Long> streamCut, int epochLow
                         ImmutableList<StreamSegmentRecord> newSegments = newSegmentsBuilder.build();
                         builder.addAll(newSegments);
                         // epoch record
-                        EpochRecord epochRecord = new EpochRecord(epochTransition.getNewEpoch(), epochTransition.getNewEpoch(), 
-                                builder.build(), time);
-
-                        HistoryTimeSeriesRecord timeSeriesRecord = 
+                        SimpleEntry<Long, Long> splitMergeCountLastEpoch = getSplitMergeCountsTillEpoch(currentEpoch).join();
+                        ImmutableList<StreamSegmentRecord> epochSegments = builder.build();
+                        List<StreamSegmentRecord> newEpochSegments = epochSegments.stream().collect(Collectors.toList());
```

Comment:
dont epoch segments capture new epoch segments? because we are building the new segment list in that builder.

Reply:
epochSegments is an ImmutableList and `findSegmentSplitsMerges` needs a List as input, hence the conversion.

Reply:
fixed